packages/drivers/odsp-driver/src/test/getVersions.spec.ts

```diff
@@ -0,0 +1,212 @@
+            async () => createResponse({ "x-fluid-epoch": "epoch1" }, odspSnapshot, 200),
+        );
+        console.log(version);
+    });
+
+    it("cache fetch succeeds and network fetch succeeds", async () => {
+        const resolved = await resolver.resolve({ url: odspUrl });
+
+        const service = new OdspDocumentStorageService(
+            resolved,
+            async (_options) => "token",
+            logger,
+            true,
+            { ...nonPersistentCache, persistedCache: epochTracker},
+            hostPolicy,
+            epochTracker,
+            async () => { return {}; },
+        );
+
+        const cacheEntry: ICacheEntry = {
+            key:"",
+            type: "snapshot",
+            file: { docId: hashedDocumentId, resolvedUrl } };
+        await localCache.put(cacheEntry, value);
+
+        const version = await mockFetchSingle(
+            async () => service.getVersions(null,1),
+            async () => createResponse({ "x-fluid-epoch": "epoch1" }, odspSnapshot, 200),
+        );
+        console.log(version);
+    });
+
+    it("cache fetch throws and network fetch throws", async () => {
+        localCache.get = async () => {
+            throw new Error("hello");
+        };
+
+        const resolved = await resolver.resolve({ url: odspUrl });
+
+        const service = new OdspDocumentStorageService(
+            resolved,
+            async (_options) => "token",
+            logger,
+            true,
+            { ...nonPersistentCache, persistedCache: epochTracker},
+            hostPolicy,
+            epochTracker,
+            async () => { return {}; },
+        );
+
+        try { await mockFetchSingle(
+            async () => service.getVersions(null,1),
+            notFound,
+        );
+        } catch (error) {
+            assert.strictEqual(error.message, "Error 404 (undefined)", "incorrect error message");
+        }
+    });
+
+    it("cache fetch succeeds and network fetch throws", async () => {
+        const resolved = await resolver.resolve({ url: odspUrl });
+
+        const service = new OdspDocumentStorageService(
+            resolved,
+            async (_options) => "token",
+            logger,
+            true,
+            { ...nonPersistentCache, persistedCache: epochTracker},
+            hostPolicy,
+            epochTracker,
+            async () => { return {}; },
+        );
+
+        const cacheEntry: ICacheEntry = {
+            key:"",
+            type: "snapshot",
+            file: { docId: hashedDocumentId, resolvedUrl } };
+        await localCache.put(cacheEntry, value);
+
+        const version = await mockFetchSingle(
+            async () => service.getVersions(null,1),
+            notFound,
+        );
+        console.log(version);
+    });
+
+    it("empty cache and network fetch throws", async () => {
+        const resolved = await resolver.resolve({ url: odspUrl });
+
+        const service = new OdspDocumentStorageService(
```

Comment:
Service looks same in all test cases, so I think we should move it to beforeEach api.